spec.html

```diff
@@ -243,8 +243,443 @@ <h1>
+          1. Set _generator_.[[GeneratorState]] to ~suspendedStart~.
+          1. Return ~unused~.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-generatorvalidate" type="abstract operation">
+        <h1>
+          GeneratorValidate (
+            _generator_: an ECMAScript language value,
+            _generatorBrand_: a String or ~empty~,
+          ): either a normal completion containing one of ~suspendedStart~, ~suspendedYield~, or ~completed~, or a throw completion
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. Perform ? RequireInternalSlot(_generator_, [[GeneratorState]]).
+          1. Perform ? RequireInternalSlot(_generator_, [[GeneratorBrand]]).
+          1. If _generator_.[[GeneratorBrand]] is not _generatorBrand_, throw a *TypeError* exception.
+          1. <del>Assert: _generator_ also has a [[GeneratorContext]] internal slot.</del>
+          1. <ins>Assert: _generator_ also has [[GeneratorContext]] and [[GeneratorAsyncContextMapping]] internal slots.</ins>
+          1. Let _state_ be _generator_.[[GeneratorState]].
+          1. If _state_ is ~executing~, throw a *TypeError* exception.
+          1. Return _state_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-generatorresume" type="abstract operation">
+        <h1>
+          GeneratorResume (
+            _generator_: an ECMAScript language value,
+            _value_: an ECMAScript language value or ~empty~,
+            _generatorBrand_: a String or ~empty~,
+          ): either a normal completion containing an ECMAScript language value or a throw completion
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. Let _state_ be ? GeneratorValidate(_generator_, _generatorBrand_).
+          1. If _state_ is ~completed~, return CreateIterResultObject(*undefined*, *true*).
+          1. Assert: _state_ is either ~suspendedStart~ or ~suspendedYield~.
+          1. Let _genContext_ be _generator_.[[GeneratorContext]].
+          1. Let _methodContext_ be the running execution context.
+          1. Suspend _methodContext_.
+          1. Set _generator_.[[GeneratorState]] to ~executing~.
+          1. <ins>Let _asyncContextMapping_ be AsyncContextSwap(_generator_.[[GeneratorAsyncContextMapping]]).</ins>
+          1. Push _genContext_ onto the execution context stack; _genContext_ is now the running execution context.
+          1. <emu-meta effects="user-code">Resume the suspended evaluation of _genContext_</emu-meta> using NormalCompletion(_value_) as the result of the operation that suspended it. Let _result_ be the value returned by the resumed computation.
+          1. Assert: When we return here, _genContext_ has already been removed from the execution context stack and _methodContext_ is the currently running execution context.
+          1. <ins>AsyncContextSwap(_asyncContextMapping_).</ins>
```

Comment:
Nit: Write this as "Perform AsyncContextSwap(_asyncContextMapping_), discarding the output."

Reply:
Actually we should probably follow this with an assert that it doesn't matter whether we are discarding or saving/restoring.